pallets/connectors-gateway/connectors-gateway-routers/src/mock.rs

```diff
@@ -0,0 +1,475 @@
+use std::str::FromStr;
+
```

Comment:
Wondering if we need a runtime to test this if it's not a pallet. Not being a pallet means that we do not need to store anything, so theoretically, these unit tests can be done without a Runtime 🤔

Reply:
Given that some of these routers actually use pallets during init/send, I think it doesn't hurt to have this.

Reply:
Then, what you're testing is the correct functionality of those pallets routers use, isn't it? Could those pallets change in the future and break the runtime mock/tests?

I am asking this from the point of view of having the minimum needed to ensure (ideally) 100% coverage of unitary testing without unneeded dependencies.  (pedantic mode on)

Reply:
> Then, what you're testing is the correct functionality of those pallets routers use, isn't it? Could those pallets change in the future and break the runtime mock/tests?
Yes, the tests also rely on the underlying pallets behaving correctly, I don't think I have a way of mocking those out or am I mistaken?

Reply:
You're right then! Sorry for the noise